### ISSUE TITLE
Incompetech song recognition slow.

### ISSUE THREAD

**Issue by tgrelka** [Timtam/incompetech-downloader#1]
When trying to download (or update) my incompetech song library, instead of skipping already exisint songs very fast (like with it does with other websites), the script is stuck at "Downloading song #" for a few seconds (~10) before the rest of the message shows up: "Parsing page for song #" etc.

**Comment by tgrelka** [comment]
Closing due to inactivity.

**Comment by Timtam** [comment]
totally forgot about that one. You're right and you cannot do anything about it, since you always need to parse the entire page to get all the required information needed to actually compare the file name against the already existing file. Other pages are much faster in that case, but incompetech itself is rather slow and I didn't find any way to speed it up yet, although I haven't looked into it for years now either. I'll have to rewrite the entire app for Python 3 as well and will probably come back to this one then too.

**Comment by tgrelka** [comment]
In the three years since opening this issue I have gained a lot of experience (as one would hope).
I work with web technology as well as parsing complex and large volumes of data on a daily basis and might be able to assist with this issue! However, it’s no longer a relevant piece of software for me (back then I was listening to lots of open-domain music, today I got Spotify 😛) and I’m not sure how willing you are to work on this. If you get around, though, I’d be open to helping.
